sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -608,6 +723,22 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
         val tz = ctx.addReferenceObj("timeZone", timeZone)
         (c, evPrim, evNull) => s"""$evPrim = UTF8String.fromString(
           org.apache.spark.sql.catalyst.util.DateTimeUtils.timestampToString($c, $tz));"""
+      case ar: ArrayType =>
+        val bufferClass = classOf[StringBuffer].getName
+        val buffer = ctx.addMutableState(bufferClass, "buffer", v => s"$v = new $bufferClass();")
```

Comment:
We are casting array to string, which is expected to be slow. We don't need to save the object creation by introducing this mutable state. We can always create a new one.

Reply:
ok